csrc/transform_view.cpp

```diff
@@ -269,27 +294,43 @@ class SplitTransform final : public ViewTransform {
 
     IterDomain* id = current_transformed_domain.at(index_);
     if (!id->isRFactorProduct()) {
-      id = replaceRootIdWithRFactor(root_domain, id);
+      // Never realize an expansion for a split, since we can always split and
+      // expanded domain into two new expanded domains
+      id = replaceRootIdWithRFactor(root_domain, id, false);
     }
 
     NVF_ERROR(
         id->start()->isZeroInt(),
         "Didn't expect to apply view transformations on an iter domain",
         " starting at a non-zero position.");
 
-    Val* remainder = ceilDiv(id->extent(), factor);
+    Val* remainder = ceilDiv(id->getMaybeExpandedExtent(), factor);
+
+    auto outer_extent = factor;
+    Val* outer_expanded_extent = nullptr;
+    auto inner_extent = remainder;
+    Val* inner_expanded_extent = nullptr;
+    if (id->hasExpandedExtent()) {
+      outer_extent = FusionGuard::getCurFusion()->oneVal();
+      outer_expanded_extent = factor;
+      inner_extent = FusionGuard::getCurFusion()->oneVal();
+      inner_expanded_extent = remainder;
+    }
```

Comment:
Instead of assigning an initial value and later maybe replacing it, I think it's clearer to just assign the right value for once. 

```suggestion
    Val* outer_extent, *outer_expanded_extent;
    Val* inner_extent, *inner_expanded_extent;
    if (id->hasExpandedExtent()) {
      outer_extent = FusionGuard::getCurFusion()->oneVal();
      outer_expanded_extent = factor;
      inner_extent = FusionGuard::getCurFusion()->oneVal();
      inner_expanded_extent = remainder;
    } else {
      outer_extent = factor;
      outer_expanded_extent = nullptr;
      inner_extent = remainder;
      inner_expanded_extent = nullptr;
    }
```